notebooks/ch12/LR_Case_Study.md

```diff
@@ -0,0 +1,792 @@
+
+
+
+But what do any of these column names mean? Looking at the documentation provided by the source of the dataset, we can create a table describing each column, sometimes referred to as a data dictionary:
+
+| Column        | Description   
+| :------------- |:-------------|
+|`'Bias'`| a column of all 1's |
+|`'Urban population (percentage)'`| the size of the urban population | 
+|`'Late births (reciprocal * 100)'`| the reciprocal of the number of births to women between 45 to 49, times 100 |
+|`'Wine consumption per capita'`| the consumption of wine per capita|
+|`'Liquor consumption per capita'`| the consumption of hard liquor per capita|
+|`'Cirrhosis death rate'`| the death rate from cirrhosis |
+
+At this point, we may want to look up words we do not understand. For example, a quick Google search reveals that cirrhosis is chronic liver disease and can be caused by chronic alcohol abuse. Some of these column names are really long, so it may be convenient to rename some of them later.
+
+Now that we have a good understanding of each of the columns, we can actually start to look at the data. Let's take a quick look at the first few rows of our dataset using the `head` function. 
+
+
+```python
+drinking_data.head()
+```
+
+
+
+
+<div>
+<style>
+    .dataframe thead tr:only-child th {
+        text-align: right;
+    }
+
+    .dataframe thead th {
+        text-align: left;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>Bias</th>
+      <th>Urban population (percentage)</th>
+      <th>Late births (reciprocal * 100)</th>
+      <th>Wine consumption per capita</th>
+      <th>Liquor consumption per capita</th>
+      <th>Cirrhosis death rate</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>0</th>
+      <td>1</td>
+      <td>44.0</td>
+      <td>33.2</td>
+      <td>5</td>
+      <td>30</td>
+      <td>41.2</td>
+    </tr>
+    <tr>
+      <th>1</th>
+      <td>1</td>
+      <td>43.0</td>
+      <td>33.8</td>
+      <td>4</td>
+      <td>41</td>
+      <td>31.7</td>
+    </tr>
+    <tr>
+      <th>2</th>
+      <td>1</td>
+      <td>48.0</td>
+      <td>40.6</td>
+      <td>3</td>
+      <td>38</td>
+      <td>39.4</td>
+    </tr>
+    <tr>
+      <th>3</th>
+      <td>1</td>
+      <td>52.0</td>
+      <td>39.2</td>
+      <td>7</td>
+      <td>48</td>
+      <td>57.5</td>
+    </tr>
+    <tr>
+      <th>4</th>
+      <td>1</td>
+      <td>71.0</td>
+      <td>45.5</td>
+      <td>11</td>
+      <td>53</td>
+      <td>74.8</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+Although we cannot see any null values from the first 5 rows of the dataset, there may be some present in our dataset that we will need to deal with. We can write code to loop through every column of the dataset and tell us if there are null values present using the pandas `isnull` function.
```

Comment:
there may be some present in our dataset that we will need to deal with -> there may still be some in our dataset

Reply:
We can do this without a for loop!

Reply:
How can we do this without a for loop? I'm a noob